packages/ui/src/lib/layouts/LayoutEditor.svelte

```diff
@@ -0,0 +1,312 @@
+<script lang="ts">
+import { faBars, faCheck, faGripVertical, faUndo } from '@fortawesome/free-solid-svg-icons';
+import { scale, SvelteMap } from 'svelte/reactivity';
+
+import { Icon } from '../icons';
+import type { LayoutEditItem } from './types';
+
+interface Props {
+  items: LayoutEditItem[];
+  ordering?: Map<string, number>;
+  title?: string;
+  enableReorder?: boolean;
+  enableToggle?: boolean;
+  onOrderChange?: (newOrdering: Map<string, number>) => void;
+  onToggle?: (itemId: string, enabled: boolean) => void;
+  onReset?: () => void;
+  resetButtonLabel?: string;
+  class?: string;
+}
+
+let {
+  items = [],
+  ordering = new Map(),
+  title = 'Select items',
+  enableReorder = false,
+  enableToggle = false,
+  onOrderChange,
+  onToggle,
+  onReset,
+  resetButtonLabel = 'Reset to default',
+  class: className = '',
+}: Props = $props();
+
+// Create ordered items list based on the ordering Map
+let orderedItems = $derived.by(() => {
+  // If no ordering provided, use original order
+  if (ordering.size === 0) {
+    return [...items].sort((a, b) => a.originalOrder - b.originalOrder);
+  }
+
+  // All items should be in the ordering map when reordering is active
+  return [...items].sort((a, b) => {
+    const aOrder = ordering.get(a.id) ?? a.originalOrder;
+    const bOrder = ordering.get(b.id) ?? b.originalOrder;
+    return aOrder - bOrder;
+  });
+});
+
+// Create a reactive set derived from items (read-only)
+let enabledItems = $derived(new Set(items.filter(item => item.enabled).map(item => item.id)));
```

Comment:
You can directly use a SvelteSet instead,


```suggestion
let enabledItems = new SvelteSet(items.filter(item => item.enabled).map(item => item.id));
```